### DISCUSSION TITLE
How to set io_manager when creating asset from op?

### DISCUSSION THREAD

**Discussion by andrey999333** [dagster-io/dagster#11277] [Q&A]
When using asset decorator i can easily set up custom io_manager by `@asset(io_manager_key="csv_saver")`, but when creating asset from op i cannot do that. `AssetsDefinition.from_op(op, io_manager_key="csv_saver")` causes an error `AssetsDefinition.from_op() got an unexpected keyword argument 'io_manager_key'`. Is there a different way to specify io_manager for the asset when creating it from an op?

Sorry, looks like i found an answer. Discussion can be deleted.

The solution is that io_manager can be set for the op. Even though this solves my problem, i think it would be nice to be able to set it up also when creating the asset.

**Comment by sryza** [comment]
Here's an issue where we're tracking adding the ability to set it when creating the asset from the op: https://github.com/dagster-io/dagster/issues/9357
